addons/binding/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/Connection.java

```diff
@@ -0,0 +1,829 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.openhab.binding.amazonechocontrol.internal;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.CookieStore;
+import java.net.HttpCookie;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.net.ssl.HttpsURLConnection;
+
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonAutomation;
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonBluetoothStates;
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonDevices;
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonDevices.Device;
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonEnabledFeeds;
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonFeed;
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonMediaState;
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonNetworkDetails;
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonNotificationRequest;
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonNotificationResponse;
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonNotificationSound;
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonNotificationSounds;
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonPlayerState;
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonPlaylists;
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonSmartHomeDevice;
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonStartRoutineRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link Connection} is responsible for the connection to the amazon server and
+ * handling of the commands
+ *
+ * @author Michael Geramb - Initial contribution
+ */
+public class Connection {
+    private final Logger logger = LoggerFactory.getLogger(Connection.class);
+
+    private java.net.CookieManager m_cookieManager = new java.net.CookieManager();
```

Comment:
Please import these classes instead of refering them via their FQN

Reply:
OK

Reply:
Thank's for your first review!!! I will change the code in the next days. Best regards to the Netherlands, Michael